### PR TITLE
Lib version

### PR DESCRIPTION
**Pull request by hzhuang1**

### CODE REVIEW

File: conf.sh [hunk 1]
```diff
@@ -14,14 +14,25 @@ if [ $1 ]; then
...
 	fi
 fi
 
+if [ $2 ]; then
+	if [ $2 = "--with-uadk_v1" ]; then
+		echo "enable UADK v1!"
+		UADK_VERSION="--with-uadk_v1"
+	fi
+fi
+
```
Comment (wangzhou) [inline conf.sh:31]: confused here. Could we add some comments about the v1/v2, static/shared lib conf.sh command?

File: configure.ac [hunk 2]
```diff
@@ -94,9 +89,19 @@ AC_FUNC_MALLOC
...
+			 test/hisi_sec_test/Makefile
+			 test/hisi_zip_test/Makefile])
+)
+
+#AC_CONFIG_FILES([Makefile
+#         test/Makefile
+#	 test/hisi_hpre_test/Makefile
+#	 test/hisi_sec_test/Makefile
+#         test/hisi_zip_test/Makefile
+#	 v1/Makefile])
```
Comment (wangzhou) [inline configure.ac:109]: can we remove above comments?

File: configure.ac [hunk 3]
```diff
@@ -94,9 +89,19 @@ AC_FUNC_MALLOC
...
-		 test/hisi_sec_test/Makefile
-                 test/hisi_zip_test/Makefile])
+AC_CONFIG_FILES([Makefile])
+AS_IF([test -n "$with_uadk_v1"],
+	AC_CONFIG_FILES([v1/Makefile]),
+	AC_CONFIG_FILES([test/Makefile
+			 test/hisi_hpre_test/Makefile
+			 test/hisi_sec_test/Makefile
+			 test/hisi_zip_test/Makefile])
+)
```
Comment (wangzhou) [inline configure.ac:102]: I am not sure if we still need this?

File: v1/test/Makefile.am [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-test_hisi_zlib_LDADD=../.libs/libwd.a
-wd_zip_test_LDADD=../.libs/libwd.a lib/libz.a
+test_hisi_zip_LDADD=../.libs/libwd.la
+test_hisi_zlib_LDADD=../.libs/libwd.la
+wd_zip_test_LDADD=../.libs/libwd.la -lz
 
 test_dummy_SOURCES=test_dummy.c
 
-test_dummy_LDADD=../.libs/libwd.a
+test_dummy_LDADD=../.libs/libwd.la
```
Comment (wangzhou) [inline v1/test/Makefile.am]: we can remove dummy test at all

File: conf.sh (hunk 1 above)
Comment (hzhuang1) [reply conf.sh:31]: OK. I'll update it with comments.

File: configure.ac (hunk 2 above)
Comment (hzhuang1) [reply configure.ac:109]: No, we can't remove them. Without them, Makefile can't be generated successfully.

File: configure.ac (hunk 3 above)
Comment (hzhuang1) [reply configure.ac:102]: Yes, we need them.

File: v1/test/Makefile.am (hunk 4 above)
Comment (hzhuang1) [reply v1/test/Makefile.am]: OK. I'll remove those dummy tests.

**Pull request closed without merging**